### PR TITLE
Aaa

### PR DESCRIPTION
**Pull request by sutaakar**
Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [ ] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Review by Sgitario — COMMENTED** [review]
As far I can see, these changes require to:
- login to the image repository first
- then run the tests using the namespace and repository name

If we use quay.io, the repository must be public before using it (and this is a manual step). 
I think this might be problematic when running the tests in a CI pipeline (as multiple tests might be pushing to the same repository name). 

What about to deploy a Docker repository inside of the Openshift test namespace? Not sure if this is something doable, but we would avoid:
- to do any login before running the tests
- add more arguments to the test script
- conflicts when running tests in parallel (by CI pipeline or by us locally)

What do you think?

File: test/steps/dockerfile_provider.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
```
Comment (Sgitario) [inline test/steps/dockerfile_provider.go]: Should this script go to the test/framework package instead?

File: test/steps/dockerfile_provider.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+type kogitoApplicationDockerfileProviderStruct struct {
+	projectLocation         string
+	baseImageName           string
+	applicationBinarySuffix string
+	libFolderNeeded         bool
+}
+
+var quarkusNonNativeKogitoApplicationDockerfileProvider = kogitoApplicationDockerfileProviderStruct{
+	baseImageName:           "quay.io/kiegroup/kogito-quarkus-jvm-ubi8",
```
Comment (Sgitario) [inline test/steps/dockerfile_provider.go]: The base image names are already in kogito_imagestream.go, concretely, we might make use of them by:
BuildImageStreams[BuildTypeRuntime][SpringbootRuntimeType] // for spring
BuildImageStreams[BuildTypeRuntime][QuarkusRuntimeType] // for quarkus native
BuildImageStreams[BuildTypeRuntimeJvm][QuarkusRuntimeType] // for quarkus non native

**Review by Sgitario — CHANGES_REQUESTED** [review]

**Comment by sutaakar** [comment]
> As far I can see, these changes require to:
> -    login to the image repository first
> -    then run the tests using the namespace and repository name

Yes.

>If we use quay.io, the repository must be public before using it (and this is a manual step).

Yes, the step is designed to be independent on registry provider, so any configuration for quay has to be done in advance before the tests are executed.

>I think this might be problematic when running the tests in a CI pipeline (as multiple tests might be pushing to the same repository name).

I would expect that registry should be able to handle pushing of several images into same repository (while images differ in image tag). Though I haven't tried that yet.

> What about to deploy a Docker repository inside of the Openshift test namespace?

For OpenShift that may be an option. For Kubernetes it can be a bit tricky as we have to add those registries to insecure registry list for docker in the node itself. May be worth investigating (and improving the implementation in future).

File: test/steps/dockerfile_provider.go (hunk 1 above)
Comment (sutaakar) [reply test/steps/dockerfile_provider.go]: well, that is a good question :)
I wasn't sure where it belongs as the line between steps and framework package is a bit blurry. But the file doesn't contain any code directly related to step, so I will move it.

File: test/steps/dockerfile_provider.go (hunk 2 above)
Comment (sutaakar) [reply test/steps/dockerfile_provider.go]: Good catch, will update it.

**Comment by sutaakar** [comment]
@Sgitario Comments addressed

**Review by Sgitario — APPROVED** [review]

**Pull request merged**